File: fe/lib/core/models/support_message.dart
class SupportMessage {
  const SupportMessage({
    required this.id,
    required this.threadId,
    required this.senderId,
    required this.senderRole,
    required this.content,
    required this.createdAt,
  });

  final String id;
  final String threadId;
  final String senderId;
  final String senderRole;
  final String content;
  final DateTime? createdAt;

  factory SupportMessage.fromJson(Map<String, dynamic> json) {
    return SupportMessage(
      id: (json['id'] ?? json['_id'] ?? '').toString(),
      threadId: (json['threadId'] ?? '').toString(),
      senderId: (json['senderId'] ?? '').toString(),
      senderRole: (json['senderRole'] ?? 'USER').toString(),
      content: (json['content'] ?? '').toString(),
      createdAt: json['createdAt'] != null
          ? DateTime.tryParse(json['createdAt'].toString())
          : null,
    );
  }
}

File: be/src/models/SupportMessage.js
const mongoose = require('mongoose');

const supportMessageSchema = new mongoose.Schema(
  {
    threadId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'SupportThread',
      required: true,
      index: true,
    },
    senderId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true,
      index: true,
    },
    senderRole: {
      type: String,
      enum: ['USER', 'ADMIN'],
      required: true,
      index: true,
    },
    content: {
      type: String,
      required: true,
      trim: true,
      maxlength: 2000,
    },
  },
  { timestamps: true },
);

supportMessageSchema.index({ threadId: 1, createdAt: 1 });

module.exports = mongoose.model('SupportMessage', supportMessageSchema);
